### PR TITLE
Refactor mobsf parser for v4 reports

### PR DESCRIPTION
**Pull request by dmarushkin**
Hi guys!

Looks like in 4th version of Mobsf json reports parser failed again for android and ios, it loose some vulns after loading and has html tags in vuln titles and descriptions. 

I've drilled down into current reports structure and parser logic and found that new reports in Mobsf have `appsec` section with same structure for ios and android reports and well rendered by maintainers vuln titles and descriptions, in mobsf UI it used in Scoreboad:

image

So I deside to refactor current parser to get dd findings and test description from this `appsec` items.

Results look like:

image

image

image

image

image

image

image

### CODE REVIEW

**Comment by Maffooch** [comment]
Does this PR maintain backward compatibility with older versions of the tool?

**Comment by dmarushkin** [comment]
@Maffooch Hi Cody!
Looks like Scoreboard and appsec section in report exists from Jan 2022 and Mobsf version 3.5 (link with [change](https://github.com/MobSF/Mobile-Security-Framework-MobSF/commit/8c10db74e10439a71f5edb2deb1bba30136d34dd))

**Comment by Maffooch** [comment]
Sorry, I should have been more specific. Is the MobSF parser compatible with older versions of the tool after the changes in this PR? It looks like all of the unit tests and sample scans are removed/rewritten in this PR. If this change to support version 4.x of MobSF cannot support older formats of MobSF, then the PR cannot be accepted

**Comment by Maffooch** [comment]
If the results are significantly different than the older format, then a new parser might be best. I'll leave that up to you to decide 😄 

The main point is that we can maintain to the old and the new formats at once

**Comment by dmarushkin** [comment]
@Maffooch Hi Cody! Sorry for long delay... I've separate all new logic into new parser MobSF Scorecard Scan and restore old parser files as is.

**Review by mtesauro — DISMISSED** [review]
Approved

**Comment by mtesauro** [comment]
@dmarushkin Please look a the failing REST tests - it appears there's  an issue with naming and the unit tests.

**Comment by dmarushkin** [comment]
@mtesauro Yeh, now messed up with paths to sample reports and lost test for old parser... 
fixed, hope that settings hash in dev has not changed again )

**Comment by mtesauro** [comment]
@dmarushkin  Yikes. Looks like that hash has changed - I went ahead and kicked off the tests again. Wait till they finish to fix the sha256sum and we should be good to start the reviews & approvals.

Thanks for your patience on this one :+1:

**Comment by dmarushkin** [comment]
@mtesauro Fixed hash, duble checked that new one will actual for new dev settings with applied change.

File: dojo/fixtures/defect_dojo_sample_data.json
Comment (Maffooch) [inline dojo/fixtures/defect_dojo_sample_data.json:1]: This file does not _need_ to be updated, but it also does not hurt to do so. I leave it up to you to back it out or not 😄

File: docs/content/en/integrations/parsers/file/mobsf_scorecard.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: "MobSF Scorecard Scanner"
+toc_hide: true
+---
+Export a JSON file using the API, api/v1/report\_json.
```
Comment (Maffooch) [inline docs/content/en/integrations/parsers/file/mobsf_scorecard.md:5]:
```suggestion
Export a JSON file using the API, api/v1/report_json.
```
Not sure if this is a typo or not

File: unittests/tools/test_mobsf_scorecard_parser.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+
+class TestMobSFScorecardParser(DojoTestCase):
+
+    def test_parse_android_empty_file(self):
+
+        test = Test()
+        engagement = Engagement()
+        engagement.product = Product()
+        test.engagement = engagement
```
Comment (Maffooch) [inline unittests/tools/test_mobsf_scorecard_parser.py:13]: This block in each function is not necessary

File: dojo/tools/mobsf_scorecard/parser.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+        tree = filename.read()
+
+        try:
+            data = json.loads(str(tree, "utf-8"))
+        except:
+            data = json.loads(tree)
+
+        if "timestamp" in data:
+            find_date = datetime.strptime(data["timestamp"], "%Y-%m-%dT%H:%M:%S.%fZ")
```
Comment (Maffooch) [inline dojo/tools/mobsf_scorecard/parser.py:28]: I have grown to be a little paranoid about relying on specific date formats. If a tool changes the date format out from under you, the parser is no longer functional.

My suggestion is use the `dateutil.parser` function ([example here](https://github.com/DefectDojo/django-DefectDojo/blob/fafe5c3cc5f1287963e4681979095b8b8064575f/dojo/tools/netsparker/parser.py#L39)), but a try/catch works in as well

**Review by Maffooch — CHANGES_REQUESTED** [review]

File: dojo/tools/mobsf_scorecard/parser.py (hunk 3 above)
Comment (dmarushkin) [reply dojo/tools/mobsf_scorecard/parser.py]: Sure thing, copy-pasted from your example, thanks for advice )

File: unittests/tools/test_mobsf_scorecard_parser.py (hunk 2 above)
Comment (dmarushkin) [reply unittests/tools/test_mobsf_scorecard_parser.py]: Removed )

File: docs/content/en/integrations/parsers/file/mobsf_scorecard.md (hunk 1 above)
Comment (dmarushkin) [reply docs/content/en/integrations/parsers/file/mobsf_scorecard.md]: Copied this typo from original parser, fixed )

File: dojo/fixtures/defect_dojo_sample_data.json
Comment (dmarushkin) [reply dojo/fixtures/defect_dojo_sample_data.json:1]: Removed changes from this file

**Review by Maffooch — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

**Review by hblankenship — APPROVED** [review]

**Review by hblankenship — APPROVED** [review]

**Review by grendel513 — APPROVED** [review]

**Comment by mtesauro** [comment]
@dmarushkin Sorry, one more fix of the sha256sum and I'll quickly merge this before it happens again.

BTW, we're looking at different methods for what that sum does so this doesn't impact contributors so sorry you've had to deal with it so much in the PR.  Hopefully, for your next contribution, it will be gone. :crossed_fingers:

**Comment by dmarushkin** [comment]
@mtesauro Not a problem, then God gave others brain, I was in queue for patience )

**Pull request merged**